buildSrc/src/main/kotlin/configureMavenSigning.kt

```diff
@@ -0,0 +1,10 @@
+import org.gradle.kotlin.dsl.get
+import org.gradle.plugins.signing.SigningExtension
+
+fun SigningExtension.addMavenCentralSigning() {
+    val gpgBase64Key = project.providers.gradleProperty("sonatypePgpKeyBase64").forUseAtConfigurationTime()
+            .orElse(project.providers.environmentVariable("SONATYPE_PGP_KEY_BASE64").forUseAtConfigurationTime())
+    val keyPassword = ""
+    useInMemoryPgpKeys(gpgBase64Key.get(), keyPassword)
+    sign(project.tasks["publishToMavenCentralRepository"])
```

Comment:
No, I just took this from the example in the official Gradle docs: https://docs.gradle.org/current/userguide/signing_plugin.html#sec:in-memory-keys

I tried to get your syntax to work, but couldn't figure it out. Feel free to change this.

Reply:
This might be related to the fact that our gradle files are in Kotlin (which has stricter type checking) rather than Groovy, but I had to use this to make it work:

```kt
sign(extensions.getByType<PublishingExtension>().publications)
```